.github/actions/push-raport-to-gh-page/action.yaml

```diff
@@ -0,0 +1,78 @@
+name: "Push raport to gh-page"
+description: "Used to push a raport file to the gh-page branch for the webiste to be updated"
+inputs:
+  artifact-name:
+    description: "Name of the artifact that has been uploaded"
+    required: true
+  file-to-pick:
+    description: "Name of the file that has been uploaded and should be picked for the raport"
+    required: true
+  build-title:
+    description: "Title of the post file that will be created"
+    required: true
+  token:
+    description: "Token for pushing"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Git Checkout
+      uses: actions/checkout@v3
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.artifact-name }}
+        path: ~/
+
+    - name: Add front matter to the file
+      shell: bash
+      run: ./scripts/insert_post_front_matter.sh ~/${{ inputs.file-to-pick }} ${{ inputs.build-title }}
+
+    - name: Git Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: gh-page
+        token: ${{ inputs.token }}
+
+    - name: Add the raport to _post dir and push
+      shell: bash
+      run: |
+        if [ -s ~/${{ inputs.file-to-pick }} ]; then
+          echo "The file is non-empty."
```

Comment:
Can you restructure that for fast exit path, eg. 
```
if [isEmpty]; then 
  warnEmpty
  exit 0
fi
```
This way we would be able to handle the erroneous state and would not need to create an else branch here.